js/src/admin/utils/ExtensionData.js

```diff
@@ -1,34 +1,159 @@
 import ItemList from '../../common/utils/ItemList';
-import listItems from '../../common/helpers/listItems';
 
 export default class ExtensionData {
-  constructor(extension) {
-    this.extension = extension;
-    app.extensionData[extension] = app.extensionData[extension] || {};
+  constructor() {
+    this.data = {};
+    this.currentExtension = null;
   }
 
-  registerData(type, content, permissionType = null, priority = 0) {
-    if (content instanceof ItemList) {
-      content = listItems(content.toArray());
-    }
+  /**
+   * This function simply takes the extension id
+   *
+   * @example
+   * app.extensionData.load('flarum-tags')
+   *
+   * flarum/flags -> flarum-flags | acme/extension -> acme-extension
+   *
+   * @param extension
+   */
+  load(extension) {
```

Comment:
Maybe it should be called `for` instead of `load`?

`app.extensionData.for('kyrne-shout').addPermission(...)`